### PR TITLE
drag and drop, host page

### PR DESCRIPTION
**Pull request by ufoBeliever**
drag and drop, host page

closing issue https://github.com/igorizviekov/mt-booking_app/issues/21

### CODE REVIEW

File: src/components/DragAndDrop/DragAndDrop.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      );
+    }
+    setDrag(false);
+  };
+
+  return (
+    <div className='flex flex-col'>
+      <div className='flex items-center dark:bg-black'>
+        <div className='flex flex-col items-center h-44 w-28 justify-between'>
+          <h4 className='text-center dark:text-white'>Drop your avatar here</h4>
```
Comment (igorizviekov) [inline src/components/DragAndDrop/DragAndDrop.tsx:59]: This should be dynamic . f.e. `label` prop

File: src/components/DragAndDrop/DragAndDrop.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+              <BsPlusLg size='2em' />
+            )}
+          </div>
+        </div>
+        {image ? (
+          <div className='flex flex-col justify-between ml-10 h-44'>
+            <img
+              src={image.toString()}
+              className='h-32 rounded'
+            />
```
Comment (igorizviekov) [inline src/components/DragAndDrop/DragAndDrop.tsx:88]: add alt prop

**Review by igorizviekov — CHANGES_REQUESTED** [review]
All above plus:

1. Clicking on drag&drop area must open the file manager
2. when I drag uploaded image onto the D&D it gives me an error: `DragAndDrop.tsx:40 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'type')` "drop here" animation freezes
3. when preview is loaded, and "Delete" btn is rendered add "replace" btn which will open a file manager
3. Add props for instructions on which file format it accepts and the max file size
4. Image preview should replace D&D area
5. add an optional prop to make this field required. make the field required on the form
6. add stories for D&D component

File: src/components/DragAndDrop/DragAndDrop.tsx (hunk 1 above)
Comment (ufoBeliever) [reply src/components/DragAndDrop/DragAndDrop.tsx]: https://github.com/igorizviekov/mt-booking_app/pull/40/commits/ac85c3c8b18d02bdf3af1b8d93764919140e815b

File: src/components/DragAndDrop/DragAndDrop.tsx (hunk 2 above)
Comment (igorizviekov) [reply src/components/DragAndDrop/DragAndDrop.tsx]: @ufoBeliever add alt prop !

File: src/components/DragAndDrop/DragAndDrop.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import { Button } from 'antd';
+import React, { useState } from 'react';
+import { BsPlusLg } from 'react-icons/bs';
+
+interface IDragAndDropProps {
+  image: string | ArrayBuffer | null;
+  setImage: (a: string | ArrayBuffer | null) => any;
+  maxFileSize: number;
```
Comment (igorizviekov) [inline src/components/DragAndDrop/DragAndDrop.tsx]: can it be refactored to not include `null` as well as `any` as a return val?

**Review by igorizviekov — APPROVED** [review]

**Pull request merged**